File: src/main/java/net/azisaba/afnw/afnwcore2/listeners/player/QuitListener.java
package net.azisaba.afnw.afnwcore2.listeners.player;

import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerQuitEvent;

/**
 * ログアウトしたプレイヤーに関するクラス
 * @author m2en
 * @see org.bukkit.event.Listener
 */
public class QuitListener implements Listener {

    /**
     * ログアウト通知を行います。
     * @param e Target player for PlayerQuitEvent
     */
    @EventHandler(priority = EventPriority.NORMAL)
    public void onQuit(PlayerQuitEvent e) {
        Player p = e.getPlayer();
        e.quitMessage(Component.text(p.getName() + "がログアウトしました。").color(NamedTextColor.YELLOW));
    }
}

File: src/main/kotlin/net/azisaba/afnw/afnwcore2/listener/player/QuitListener.kt
package net.azisaba.afnw.afnwcore2.listener.player

import com.earth2me.essentials.Essentials
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor
import org.bukkit.Bukkit
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerQuitEvent

class QuitListener: Listener {

    @EventHandler
    fun onQuit(event: PlayerQuitEvent) {
        val player: Player = event.player

        if(isVanished(player)) {
            return
        }

        event.quitMessage(Component.text("* " + player.name + " がログアウトしました", NamedTextColor.AQUA))
    }

    private fun isVanished(player: Player): Boolean {
        val essentials = Bukkit.getPluginManager().getPlugin("Essentials") as Essentials?
            ?: return false
        return essentials.getUser(player).isVanished
    }
}
